src/client/app/components/csv/ReadingsCSVUploadComponent.tsx

```diff
@@ -26,315 +33,477 @@ function range(lower: number, upper: number): number[] {
-						</Col>
-					</FormGroup>
-					<FormGroup>
-						<Label style={titleStyle}>
-							<FormattedMessage id='csv.readings.param.time.sort' />
-						</Label>
-						<Col sm={8}>
-							<Input type='select' name='timeSort' onChange={this.handleSetTimeSort}>
-								<option value={TimeSortTypes.meter}> {translate('TimeSortTypes.meter')} </option>
-								<option value={TimeSortTypes.increasing}> {translate('TimeSortTypes.increasing')} </option>
-								<option value={TimeSortTypes.decreasing}> {translate('TimeSortTypes.decreasing')} </option>
-							</Input>
-						</Col>
-					</FormGroup>
-					<FormGroup>
-						<Label style={titleStyle}>
-							<FormattedMessage id='csv.readings.param.duplications' />
-						</Label>
-						<Col sm={8}>
-							<Input value={this.props.duplications} type='select' name='duplications' onChange={this.handleSetDuplications}>
-								{range(1, 10).map(i => (
-									<option key={i} value={`${i}`}> {i} </option>
-								))}
+	return (
+		<Container>
+			<TooltipHelpComponent page='help.csv.header' />
+			<Form onSubmit={handleSubmit}>
+				<Row className="justify-content-md-center">
+					<Col md='auto'>
+						<h2>
+							{translate('csv.upload.readings')}
+							<div style={tooltipStyle}>
+								<TooltipMarkerComponent page='help.csv.header' helpTextId={tooltipStyle.tooltipReadings} />
+							</div>
+						</h2>
+					</Col>
+				</Row>
+				<Row className='justify-content-md-center'>
+					<Col md='auto'>
+						<Label for='meterName'>
+							<div className='pb-1'>
+								{translate('name')}
+							</div>
+							<Input
+								id='meterName'
+								name='meterName'
+								type='select'
+								value={selectedMeter?.name || ''}
+								onChange={handleSelectedMeterChange}>
+								<option value='default' key='-1'>Select a Meter</option>
+								{Array.from(visibleMeters).map(meter => {
+									return (<option value={meter.name} key={meter.id}>{meter.name}</option>);
+								})}
 							</Input>
-						</Col>
-					</FormGroup>
-					<FormFileUploaderComponent formText='csv.upload.readings' reference={this.fileInput} required labelStyle={titleStyle} />
-					<FormGroup>
-						<Label style={titleStyle}>
-							<FormattedMessage id='csv.readings.section.cumulative.data' />
 						</Label>
-						<Col sm={8}>
-							<FormGroup>
-								<Label style={titleStyle}>
-									<FormattedMessage id='csv.readings.param.cumulative' />
-								</Label>
-								<Col sm={12}>
-									<Input type='select' name='cumulative' onChange={this.handleSetCumulative}>
-										<option value={BooleanMeterTypes.meter}> {translate('BooleanMeterTypes.meter')} </option>
-										<option value={BooleanMeterTypes.true}> {translate('BooleanMeterTypes.true')} </option>
-										<option value={BooleanMeterTypes.false}> {translate('BooleanMeterTypes.false')} </option>
-									</Input>
+						<br />
+						<CreateMeterModalComponent onCreateMeter={handleCreateMeter}/>
```

Comment:
I tried being logged in with the CSV role and was able to create a new meter. However, only an admin should be able to do that and if the CSV user goes to the meter page they cannot. I'm not sure what is up but it is an issue.

Also, if a CSV user cannot create a meter then it might be good to revisit the logic above about a newly created meter that is not visible to the CSV user.